transcribe/scribe_modules/yum_repos.py

```diff
@@ -16,19 +18,17 @@ def __init__(self, input_dict=None, module_name=None, host_name=None,
                                        input_type=input_type,
                                        scribe_uuid=scribe_uuid)
         if input_dict:
-            # object_dict['repo_name'] = input_dict['repoid']
-            # object_dict['repo_state'] = self.update_repo_state(input_dict)
-            # object_dict['base_url'] = format_url(base_url, self.repo_name)
-            self.repo_name = input_dict['repoid']
-            self.repo_state = self.update_repo_state(input_dict)
-            # This is just for the sake of it
-            self.base_url = format_url(base_url, self.repo_name)
-
-    def update_repo_state(self, value):
-        if value['state'] == 'enabled':
-            return 1
-        return 0
+            self._input_dict = input_dict
+        else:
+            raise ValueError('Input dictionary is empty')
```

Comment:
This pattern is repeated so many times across all the modules.  Seems like it is worth to follow the DRY principle here and find a way to abstract that away it is done the same way, once, and done correctly.

Reply:
Agreed, i'll just create a simple function in a lib and use that